plugins/protocolbuffers/kotlin/v24.4/escape-package-names.patch

```diff
@@ -0,0 +1,64 @@
+diff --git a/src/google/protobuf/compiler/java/message.cc b/src/google/protobuf/compiler/java/message.cc
+index 3ef504526..3d46c2e43 100644
+--- a/src/google/protobuf/compiler/java/message.cc
++++ b/src/google/protobuf/compiler/java/message.cc
+@@ -1337,8 +1337,7 @@ void ImmutableMessageGenerator::GenerateTopLevelKotlinMembers(
+       "message",
+       EscapeKotlinKeywords(name_resolver_->GetClassName(descriptor_, true)),
+       "message_kt",
+-      EscapeKotlinKeywords(
+-          name_resolver_->GetKotlinExtensionsClassName(descriptor_)));
++      name_resolver_->GetKotlinExtensionsClassNameEscaped(descriptor_));
+ 
+   for (int i = 0; i < descriptor_->nested_type_count(); i++) {
+     if (IsMapEntry(descriptor_->nested_type(i))) continue;
+diff --git a/src/google/protobuf/compiler/java/message_lite.cc b/src/google/protobuf/compiler/java/message_lite.cc
+index b5fddc2a9..7321ed31f 100644
+--- a/src/google/protobuf/compiler/java/message_lite.cc
++++ b/src/google/protobuf/compiler/java/message_lite.cc
+@@ -859,8 +859,7 @@ void ImmutableMessageLiteGenerator::GenerateTopLevelKotlinMembers(
+       "message",
+       EscapeKotlinKeywords(name_resolver_->GetClassName(descriptor_, true)),
+       "message_kt",
+-      EscapeKotlinKeywords(
+-          name_resolver_->GetKotlinExtensionsClassName(descriptor_)));
++      name_resolver_->GetKotlinExtensionsClassNameEscaped(descriptor_));
+ 
+   for (int i = 0; i < descriptor_->nested_type_count(); i++) {
+     if (IsMapEntry(descriptor_->nested_type(i))) continue;
+diff --git a/src/google/protobuf/compiler/java/name_resolver.cc b/src/google/protobuf/compiler/java/name_resolver.cc
+index 0017e172a..b027865b1 100644
+--- a/src/google/protobuf/compiler/java/name_resolver.cc
++++ b/src/google/protobuf/compiler/java/name_resolver.cc
+@@ -358,6 +358,19 @@ std::string ClassNameResolver::GetKotlinExtensionsClassName(
+                           descriptor->file(), true, true, true);
+ }
+ 
++std::string ClassNameResolver::GetKotlinExtensionsClassNameEscaped(
```

Comment:
I can't remember why we have this patch. I wish we kept a list (or commented) why these exist and what specific problem they solve. And if there's any upstream issue we're tracking.